utils/check_roxie/check_roxie

```diff
@@ -1,11 +1,74 @@
-#!/bin/bash
+#!/usr/bin/env python
 
-OUTPUT=`testsocket $1 "<control:alive/>" |grep Status |grep ok -c`
+from __future__ import print_function
+import subprocess
+import re
+import sys
 
-if [ "$OUTPUT" -eq "1" ]; then
-    echo "OK"
-    exit 0
-else
-    echo "CRITICAL"
-    exit 2
-fi
+# Output usage if argument is missing
+try:
+    IP = sys.argv[ 1 ]
+except IndexError:
+    print ( "\nNeeds IP address of the target server" )
+    print ( """
+Usage:
+        check_roxie <ip-address>
+""" )
+    sys.exit ( 1 )
+
+# Nagios exit codes
+ok = 0
+warning = 1
+critical = 2
+
+cmd_roxie_status = '2>/dev/null testsocket %s "<control:state/>"' % IP
+status = subprocess.Popen ( cmd_roxie_status.strip ( ) , stdout=subprocess.PIPE , shell=True )
+status_result = status.communicate ( )[ 0 ]
+roxie_status = re.findall ( '\<Status\>(.+?)\</Status\>+?' , status_result )
+my_hash = re.findall ( "hash='(.*?)'" , status_result )
+
+# Check if roxie process is up and running
+if roxie_status and roxie_status[ 0 ] == 'ok':
+    # Get state hashes from roxie cluster
+    hashes = [ ]
+    try:
+        cmd_cluster_state = '2>/dev/null testsocket %s -lock "<control:state/>"' % IP
+        cluster_state = subprocess.Popen ( cmd_cluster_state.strip ( ) , stdout=subprocess.PIPE , shell=True )
+        cluster_state_result = cluster_state.communicate ( )[ 0 ]
+        for cluster_result in cluster_state_result.split():
+            if "hash" in cluster_result:
+                state_hash = re.findall ( "hash=\"(.*?)\"" , cluster_result )
+                hashes.append ( state_hash[ 0 ] )
+    except:
+        pass
+
+    total_hashes = len ( hashes )
+    same_hash = [ ]
+    different_hash = [ ]
+
+    # Compare hash state between self and cluster
+    while total_hashes > 0:
+        for i in hashes:
+            if my_hash[ 0 ] == i:
+                same_hash.append ( i )
```

Comment:
I am still not convinced about the the necessity of collecting same and different hashes into arrays, when only their counts rather than their values are used in your code. 
The "my_hash[ 0 ] in same_hash" check in you "it...elif.." chain already done in the for loop, so I think:
```
myHashMatchWithOthers = False
numOfSameHashes = 0
numOfDifferentHashes = 0

for i in hashhes:
        if my_hash[ 0 ] == i:
                myHashMatchWithOthers = True
                numOfSameHashes += 1
            elif my_hash[ 0 ] != i:
                numOfDifferentHashes += 1

if numOfSameHashes < numOfDifferentHashes and myHashMatchWithOthers:
   . . .
```
Or something like that. That would be a bit simpler and doesn't use unnecessary arrays and array operations.